resources/services/sagemaker/training_jobs.go

```diff
@@ -751,8 +765,7 @@ func resolveSagemakerTrainingJobProfilerConfig(_ context.Context, _ schema.Clien
 	r, ok := resource.Item.(*sagemaker.DescribeTrainingJobOutput)
 	if !ok {
 		return fmt.Errorf("expected DescribeTrainingJobOutput but got %T", resource.Item)
-	}
-	if r.ProfilerConfig == nil {
+	} else if r == nil || r.ProfilerConfig == nil {
```

Comment:
nit: I suggest this instead of the else if for all the changes here
```suggestion
	if r != nil && r.ProfilerConfig == nil {
```

Reply:
This wouldn't cover all cases, e.g. `r` being `nil`.